core/src/main/java/org/infinispan/CacheStream.java

```diff
@@ -291,4 +341,297 @@
     */
    @Override
    <R1, A> R1 collect(Collector<? super R, A, R1> collector);
+
+   /**
+    * Same as {@link CacheStream#collect(Supplier, BiConsumer, BiConsumer)}  except that the various arguments must
+    * also be Serializable.
+    * <p>
+    * This method will be used automatically by lambdas, which prevents users from having to manually cast to
+    * a Serializable lambda.
+    * @param <R1> type of the result
+    * @param supplier a function that creates a new result container. For a
+    *                 parallel execution, this function may be called
+    *                 multiple times and must return a fresh value each time.
+    *                 Must be serializable
+    * @param accumulator an associative, non-interfering, stateless
+    *                    function for incorporating an additional element into a result and
+    *                    must be serializable
+    * @param combiner an associative, non-interfering, stateless
+    *                    function for combining two values, which must be
+    *                    compatible with the accumulator function and serializable
+    * @return the result of the reduction
+    */
+   <R1> R1 collect(SerializableSupplier<R1> supplier, SerializableBiConsumer<R1, ? super R> accumulator,
+           SerializableBiConsumer<R1, R1> combiner);
+
+   /**
+    * Same as {@link CacheStream#allMatch(Predicate)} except that the Predicate must also implement Serializable.
+    * <p>
+    * This method will be used automatically by lambdas, which prevents users from having to manually cast to
+    * a Serializable lambda.
+    * @param predicate a non-interfering, stateless
+    *                  predicate to apply to elements of this stream that is serializable
+    * @return {@code true} if either all elements of the stream match the
+    * provided predicate or the stream is empty, otherwise {@code false}
+    */
+   boolean allMatch(SerializablePredicate<? super R> predicate);
```

Comment:
Ah gotcha, yeah we may have to bring them all up some day just for Javadoc reasons, heh.